src/Text/Markdown/SlamDown/Syntax/Value.purs

```diff
@@ -1,16 +1,83 @@
 module Text.Markdown.SlamDown.Syntax.Value
   ( Value
   , stringValue
+  , decimalValue
+  , timeValue
+  , dateValue
+  , dateTimeValue
+  , thunkValue
   , renderValue
+
+  , TimeValue()
+  , DateValue()
   ) where
 
 import Prelude
+import Data.Array as A
+import Data.Maybe as M
+import Data.String as S
+import Data.HugeNum as HN
+import Text.Markdown.SlamDown.Syntax.TextBoxType
+
+type TimeValue =
+  { hours :: Int
+  , minutes :: Int
+  }
+
+type DateValue =
+  { month :: Int
+  , day :: Int
+  , year :: Int
+  }
 
 class (Eq a) <= Value a where
   stringValue :: String -> a
+  decimalValue :: HN.HugeNum -> a
+  timeValue :: TimeValue -> a
+  dateValue :: DateValue -> a
+  dateTimeValue :: DateValue -> TimeValue -> a
+
+  -- ???
+  thunkValue :: String -> a
```

Comment:
This is a bizarre thing that I added temporarily in order to be able to port the evaluator, but we need to find a way to get rid of it, since I have no idea what it means.